org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/utils/XMLGenerator.java

```diff
@@ -226,12 +227,15 @@ public static String generateAttributeValue(String defaultValue, Collection<Stri
 	 * @return
 	 */
 	public static String generateDocumentation(String documentation, String schemaURI, boolean html) {
-		StringBuilder doc = new StringBuilder(documentation != null ? documentation : "");
+		if (StringUtils.isEmpty(documentation) || StringUtils.isEmpty(documentation.trim())) {

```

Comment:
Calling` documentation.trim()` will create a new instance of String which is shame just to check that documentation contains just whitespaces. 

Please try:

`if (StringUtils.isEmpty(documentation) || StringUtils.isWhitespace(documentation)) {`

Reply:
I added a isBlank function to StringUtils that checks `StringUtils.isEmpty(documentation) || StringUtils.isWhitespace(documentation)` and used this in both locations.